### PR TITLE
Add log_sink_console specs

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Add log_sink_console specs

### CODE REVIEW

File: v2/devdoc/log_sink_console_requirements.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+
+`log_sink_console.log_sink_log` implements logging to console.
+
+If `message_format` is `NULL`, `log_sink_console.log_sink_log` shall return.
+
+`log_sink_console.log_sink_log` shall obtain the time by calling `time`.
+
+`log_sink_console.log_sink_log` shall convert the time to string by calling `ctime`.
```
Comment (cyrus-jackson) [inline v2/devdoc/log_sink_console_requirements.md:26]: How does the formatted time look like?

File: v2/devdoc/log_sink_console_requirements.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+`log_sink_console.log_sink_log` shall print a line in the format:
+
+`{log_level} Time: {formatted time} File:{file}:{line} Func:{func} {optional context information} {formatted message}`
+
+`log_sink_console.log_sink_log` shall print the line in such a way that only one `printf` call is made in order to not break the line in multiple parts when displayed on the console.
```
Comment (cyrus-jackson) [inline v2/devdoc/log_sink_console_requirements.md:32]: Multiple lines are delimited by a `.`?

**Review by cyrus-jackson — APPROVED** [review]

File: v2/devdoc/log_sink_console_requirements.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+`log_sink_console.log_sink_log` shall color the lines using ANSI color codes (https://en.wikipedia.org/wiki/ANSI_escape_code#Colors), as follows:
+
+  `LOG_LEVEL_CRITICAL` shall be displayed with bright red `\x1b[31;1m`.
+
+  `LOG_LEVEL_ERROR` shall be displayed with red `\x1b[31m`.
+
+  `LOG_LEVEL_WARNING` shall be displayed with bright yellow `\x1b[33;1m`.
+
+  `LOG_LEVEL_INFO` shall be displayed with yellow `\x1b[33m`.
```
Comment (cyrus-jackson) [inline v2/devdoc/log_sink_console_requirements.md:42]: Yellow is interesting.

File: v2/devdoc/log_sink_console_requirements.md (hunk 2 above)
Comment (cyrus-jackson) [reply v2/devdoc/log_sink_console_requirements.md:32]: Oh I understood it wrongly. Makes sense to print just one line.

File: v2/devdoc/log_sink_console_requirements.md (hunk 3 above)
Comment (dcristoloveanu) [reply v2/devdoc/log_sink_console_requirements.md:42]: :-)

File: v2/devdoc/log_sink_console_requirements.md (hunk 1 above)
Comment (dcristoloveanu) [reply v2/devdoc/log_sink_console_requirements.md:26]: ctime calls asctime and per the standard it does:

The asctime function converts the broken-down time in the structure pointed to by
timeptr into a string in the form
Sun Sep 16 01:03:52 1973\n\0

**Pull request merged**